web/packages/teleport/src/lib/polyfillRandomUuid.js

```diff
@@ -0,0 +1,153 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+if (typeof crypto?.randomUUID != 'function') {
+  window.crypto.randomUUID = randomUUID;
+}
+
+// Code based on Node.js' `lib/internal/crypto/random.js`, subject
+// to Node.js license found at:
+// https://raw.githubusercontent.com/nodejs/node/master/LICENSE
```

Comment:
I'm guessing our license up top is just for lines 17-20 and then the rest is under the license present from Node.js. Should we have two in the same file? I feel like we can do that, but just double checking

Reply:
Correct, our license is just for those few lines. I'm also not sure what the correct approach is for licenses in instances like these, I can split it up into separate files if needed. What do you think @zmb3?

Reply:
Received confirmation from @zmb3 that this approach is fine.